src/flattenizer.js

```diff
@@ -16,35 +16,31 @@
  * @public
  */
 export const flatten = (unflattened, delimiter = '.') => {
-    if (typeof unflattened === 'undefined')
-        return {};
-
-    if (typeof unflattened !== 'object') {
-        throw new TypeError('unflattened is not an object');
-    }
-
-    if(typeof delimiter !== 'string') {
-        throw new TypeError('delimiter must be a string');
-    }
-
-    let flattened = {};
-
-    for (let key in unflattened) {
-        if (!unflattened.hasOwnProperty(key))
-            continue;
-
-        if (typeof unflattened[key] === 'object') {
-            let flatObject = flatten(unflattened[key], delimiter);
-            for (let subKey in flatObject) {
-                // append to create new key value and assign it's value
-                flattened[`${key}${delimiter}${subKey}`] = flatObject[subKey];
-            }
-        } else {
-            flattened[key] = unflattened[key];
-        }
+  if (typeof unflattened === 'undefined' || unflattened === null) return {};
```

Comment:
Why not return unflattened instead of empty object? 😆

Reply:
Ha, a serious comment. Good call!